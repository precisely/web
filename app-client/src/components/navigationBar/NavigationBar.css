.navbar {
    letter-spacing: -1px;
    background-color: white;
}

#brand-logo {
    width: 10em;
}

@media screen and (min-width: 992px) {
    .navbar {
       padding: 8px 300px 8px 300px;
    }
}
