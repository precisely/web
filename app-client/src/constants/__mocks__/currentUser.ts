/*
 * Copyright (c) 2011-Present, Precise.ly, Inc.
 * All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or
 * without modification, are not permitted.
 */
function mockCurrentUser() {
  return {
    isAuthenticated: jest.fn(),
    __mockisAuthenticatedSuccessCase() {
      currentUser.isAuthenticated.mockReturnValue(true);
    },
    __mockisAuthenticatedFailureCase() {
      currentUser.isAuthenticated.mockReturnValue(false);
    },

    logout: jest.fn(),
    __mockLogoutSuccessCase() {
      currentUser.logout.mockReturnValueOnce(true);
    },
    __mockLogoutFailureCase() {
      currentUser.logout.mockReturnValueOnce(false);
    },
    __resetCurrentUserMocks: () => (currentUser = mockCurrentUser())
  };
}

export let currentUser = mockCurrentUser();
