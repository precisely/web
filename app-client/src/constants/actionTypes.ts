/*
 * Copyright (c) 2017-Present, CauseCode Technologies Pvt Ltd, India.
 * All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or
 * without modification, are not permitted.
 */

// All the constants for action types should be written here
export const FETCH_DATA: string = 'containers/Demo/FETCH_DATA';

export const SET_LENGTH: string = 'containers/Demo/SET_LENGTH';
