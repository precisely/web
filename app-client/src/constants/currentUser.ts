import {AWSUser} from 'src/utils/AWSUser.ts';

// tslint:disable-next-line
export const currentUser = new AWSUser();
