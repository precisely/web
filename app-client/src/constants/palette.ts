/*
 * Copyright (c) 2017-Present, CauseCode Technologies Pvt Ltd, India.
 * All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or
 * without modification, are not permitted.
 */

// All the global CSS styles and color constants should be written here
