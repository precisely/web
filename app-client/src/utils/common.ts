/*
 * Copyright (c) 2011-Present, Precise.ly, Inc.
 * All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or
 * without modification, are not permitted.
 */

export function isEmpty(object: Object) {
  return (!object || !Object.keys(object).length);
}

export function getEnvironment() {
  return process.env.NODE_ENV || '';
}
