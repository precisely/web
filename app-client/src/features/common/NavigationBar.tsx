/*
 * Copyright (c) 2011-Present, Precise.ly, Inc.
 * All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or
 * without modification, are not permitted.
 */

import * as React from 'react';
import * as Radium from 'radium';
import {currentUser} from 'src/constants/currentUser';
import {CSS} from 'src/interfaces';
import {
  Collapse,
  Navbar,
  NavbarToggler,
  NavbarBrand,
  Nav,
  NavItem,
  NavLink,
} from 'src/features/common/ReusableComponents';

const logo = require('src/assets/precisely-logo.png');

export interface NavigationBarState {
  isOpen?: boolean;
  backgroundColor?: string;
}

@Radium
export class NavigationBar extends React.Component {
  state = {
    isOpen: false,
    backgroundColor: 'transparent',
  };

  toggle = (): void => {
    this.setState({isOpen: !this.state.isOpen});
  }

  handleClick = (): void => {
    if (currentUser.isAuthenticated()) {
      currentUser.logout();
    } else {
      currentUser.showLogin();
    }
  }

  componentDidMount(): void {
    window.addEventListener('scroll', this.handleScroll);
  }

  handleScroll = (): void => {
    this.setState({backgroundColor: window.scrollY > 50 ? 'white' : 'transparent'});
  }

  render() {
    const {isOpen, backgroundColor} = this.state;

    navBar.backgroundColor = backgroundColor;

    return (
      <Navbar light sticky="top" expand="md" toggleable="md" className="navbar" style={navBar}>
        <NavbarBrand href="/">
          <img id="brand-logo" src={logo} alt="precise.ly" style={logoStyle} />
        </NavbarBrand>
        <NavbarToggler className="navbar-toggler-right" onClick={this.toggle} />
        <Collapse isOpen={isOpen} navbar>
          <Nav className="ml-auto" navbar>
            <NavItem className="pr-4">
              <NavLink href="/about-us">ABOUT US</NavLink>
            </NavItem>
            <NavItem>
              <NavLink style={{cursor: 'pointer'}} onClick={(): void => this.handleClick()}>
                {currentUser.isAuthenticated() ? 'LOG OUT' : 'LOG IN'}
              </NavLink>
            </NavItem>
          </Nav>
        </Collapse>
      </Navbar>
    );
  }
}

const logoStyle: CSS = {
  width: '10em',
};

const navBar: CSS = {
  letterSpacing: '-1px',
  transition: 'background-color 0.4s ease',
  '@media screen and (min-width: 992px)': {
    padding: '8px 245px',
  },
};
