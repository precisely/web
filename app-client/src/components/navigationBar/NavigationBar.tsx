/*
 * Copyright (c) 2011-Present, Precise.ly, Inc.
 * All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or
 * without modification, are not permitted.
 */

import * as React from 'react';
import * as Radium from 'radium';
import {RouteComponentProps} from 'react-router';
import {logOut, isLoggedIn} from 'src/utils/cognito';
import {
    Collapse,
    Navbar,
    NavbarToggler,
    NavbarBrand,
    Nav,
    NavItem,
    NavLink,
} from 'src/components/ReusableComponents';
import 'src/components/navigationBar/NavigationBar.css';

const logo = require('src/assets/logo-horizontal.png');

export interface INavigationBarState {
    isOpen: boolean;
}

@Radium
export class NavigationBar extends React.Component<RouteComponentProps<void>, INavigationBarState> {
    state = {
        isOpen: false
    };

    toggle = (): void => {
        this.setState({
            isOpen: !this.state.isOpen
        });
    }

    handleClick = (loggedIn: boolean): void => {
        if (loggedIn) {
            logOut();
            this.props.history.replace('/');
        } else {
            this.props.history.push('/login');
        }
    }

    render() {
        const loggedIn: boolean = isLoggedIn();
        
        return (
            <Navbar light sticky="top" expand="md" toggleable="md" className="navbar">
              <NavbarBrand href="/"><img src={logo} alt="precise.ly" /></NavbarBrand>
              <NavbarToggler className="navbar-toggler-right" onClick={this.toggle} />
              <Collapse isOpen={this.state.isOpen} navbar>
                <Nav className="ml-auto" navbar>
                  <NavItem className="pr-4">
                    <NavLink href="/components/">ABOUT US</NavLink>
                  </NavItem>
                  <NavItem>
                    <NavLink style={{cursor: 'pointer'}} onClick={() => this.handleClick(loggedIn)}>
                      {loggedIn ? 'LOG OUT' : 'LOG IN'}
                    </NavLink>
                  </NavItem>
                </Nav>
              </Collapse>
            </Navbar>
        );
    }
}
