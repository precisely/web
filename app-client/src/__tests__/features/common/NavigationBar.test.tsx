jest.mock('src/constants/currentUser');

import * as React from 'react';
import * as Radium from 'radium';
import * as Adapter from 'enzyme-adapter-react-16';
import * as renderer from 'react-test-renderer';
import {ShallowWrapper,shallow,configure} from 'enzyme';
import {NavigationBar} from 'src/features/common/NavigationBar';
import {currentUser} from 'src/constants/currentUser';

configure({ adapter: new Adapter() });
Radium.TestMode.enable();

describe('Login tests Snapshot Testing :', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<NavigationBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Login Test After Login: ', () => {
  const componentTree: ShallowWrapper = shallow(<NavigationBar />);
  it('should render Log Out link in the navigation bar when the user is authenticated.', () => {
    currentUser[`__mockisAuthenticatedSuccessCase`]();
    componentTree.find('#loginStatus').simulate('click');
    expect(
      componentTree
        .find('#loginStatus')
        .children()
        .text()
    ).toEqual('LOG IN');
  });

  it('should render Log In link in the navigation bar when the user is unauthenticated.', () => {
    currentUser[`__mockisAuthenticatedSuccessCase`]();
    componentTree.find('#loginStatus').simulate('click');
    expect(
      componentTree
        .find('#loginStatus')
        .children()
        .text()
    ).toEqual('LOG IN');
  });

  it(
    'should render Log Out link in the navigation bar when the user is authenticated after toggle button is clicked',
    () => {
    currentUser[`__mockisAuthenticatedSuccessCase`]();
    componentTree.find('.navbar-toggler-right').simulate('click');
    expect(
      componentTree
        .find('#loginStatus')
        .children()
        .text()
    ).toEqual('LOG OUT');
  });
});

describe('Login Test After Logout: ', () => {
  const componentTree: ShallowWrapper = shallow(<NavigationBar />);
  it('should render Log In link in the navigation bar when the user is unauthenticated.', () => {
    currentUser[`__mockLogoutFailureCase`]();
    componentTree.find('#loginStatus').simulate('click');
    expect(
      componentTree
        .find('#loginStatus')
        .children()
        .text()
    ).toEqual('LOG IN');
  });
});
