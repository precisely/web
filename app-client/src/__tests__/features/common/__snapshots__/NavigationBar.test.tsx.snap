// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar tests Snapshot Testing : renders correctly 1`] = `
<nav
  className="rmq-4a0d0fdc navbar navbar navbar-toggleable-md navbar-light sticky-top"
  data-radium={true}
  expand="md"
  style={
    Object {
      "backgroundColor": "transparent",
      "letterSpacing": "-1px",
      "transition": "background-color 0.4s ease",
    }
  }
>
  <a
    className="navbar-brand"
    data-radium={true}
    href="/"
  >
    <img
      alt="precise.ly"
      data-radium={true}
      id="brand-logo"
      src="precisely-logo.png"
      style={
        Object {
          "width": "10em",
        }
      }
    />
  </a>
  <button
    className="navbar-toggler-right navbar-toggler"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    data-radium={true}
    style={Object {}}
  >
    <ul
      className="ml-auto navbar-nav"
      data-radium={true}
    >
      <li
        className="pr-4 nav-item"
      >
        <a
          className="nav-link"
          href="/about-us"
          onClick={[Function]}
        >
          ABOUT US
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          data-radium={true}
          id="loginStatus"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          LOG IN
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
